async-opcua-client/src/session/services/subscriptions/event_loop.rs

```diff
@@ -162,16 +168,13 @@ impl SubscriptionEventLoop {
                                             // If this happens we will probably eventually fail keep-alive, defer to that.
                                             session_error!(slf.session, "Publish response indicates session is dead");
                                         }
-                                        StatusCode::BadNoSubscription
-                                        | StatusCode::BadSubscriptionIdInvalid => {
-                                            // TODO: Maybe do something here? This could happen when
-                                            // subscriptions are in the process of being recreated.
-                                            // Make sure to avoid race conditions.
-                                            session_error!(
+                                        StatusCode::BadNoSubscription => {
+                                            session_debug!(
                                                 slf.session,
-                                                "Publish response indicates subscription is dead",
+                                                "Publish response indicates that there are no subscriptions. Pausing publish loop until a new subscription is created, or we receive a successful publish response.",
```

Comment:
This would be a very long wrapping logline in the logfile, I prefer to keep these a bit more concise:

```suggestion
                                                "Publish response indicates that there are no subscriptions",
```